imagorpath/params_test.go

```diff
@@ -66,6 +67,31 @@ func TestParseGenerate(t *testing.T) {
 				Filters:       []Filter{{Name: "some_filter"}},
 			},
 		},
+		{
+			name:                        "non url image base64url encoded",
+			uri:                         "meta/trim/10x11:12x13/fit-in/-300x-200/left/top/smart/filters:some_filter()/b64:bXkgaW1hZ2UgKHNwZWNpYWwpPy5qcGc",
+			skipUriGenerationFromParams: true, // the reverse direction from params to path will not return the base64url encoded image; so that's why we need to disable it
```

Comment:
The reverse can be implemented by adding a field to the param struct to denote it is base64 encoded. Say `Base64Image bool`. The flag should be set true if successfully parsed. The reverse function can construct the "b64:..." path based on that flag too.

Keeping the path and the struct keeps its idempotence is useful for debugging. It's also necessary for auto injected operations such as auto webp.

Reply:
that's much more elegant :) will change this